sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):
+    def __init__(self, **opts):
+        super().__init__(**opts)
+
+    def send_mail(self, msg):
+        try:
+            return self.client.mail.send.post(request_body=msg.get())
+        except Exception as e:

```

Comment:
@thinkingserious We can do that by using the `type()` method. For Ex:-
```
try:
    raise ValueError("something")
except Exception as e:
    if type(e) is type(ValueError()):
          print("Value Error")          # 'Value Error' is shown in the output
```
Hence, by catching `Exception` in general, we can still handle it specifically according to the error.
For catching an HTTPError, we can follow a similar approach. 

What are your thoughts on this?